enterprise/api/v1/views/enterprise_group.py

```diff
@@ -37,13 +46,158 @@ def get_queryset(self, **kwargs):
+        """
+        return super().create(request, *args, **kwargs)
+
+    @action(methods=['post'], detail=False, permission_classes=[permissions.IsAuthenticated])
+    @permission_required(
+        'enterprise.can_access_admin_dashboard',
+        fn=lambda request, group_uuid: get_enterprise_customer_from_enterprise_group_id(group_uuid)
+    )
+    def assign_learners(self, request, group_uuid):
+        """
+        POST /enterprise/api/v1/enterprise-group/<group uuid>/assign_learners
+
+        Required Arguments:
+            - ``learner_emails``:
+                List of learner emails to associate with the group. Note: only processes the first 1000 records
+                provided
+
+        Returns:
+            ``records_processed``:
+                Total number of group membership records processed.
+            ``new_learners``:
+                Total number of group membership records associated with new pending enterprise learners that were
+                processed
+            ``existing_learners``:
+                Total number of group membership records associated with existing enterprise learners that were
+                processed
+        """
+        try:
+            group = self.get_queryset().get(uuid=group_uuid)
+        except models.EnterpriseGroup.DoesNotExist as exc:
+            raise Http404 from exc
+        if requested_emails := request.POST.dict().get('learner_emails'):
+            # IDK if this is necessary? bulk_create already is already a transaction that will rollback if needed.
+            with transaction.atomic():
+                total_records_processed = 0
+                total_existing_users_processed = 0
+                total_new_users_processed = 0
+                for user_email_batch in utils.batch(requested_emails.split(',')[: 1000], batch_size=200):
+                    new_pending_learners = []
+                    enterprise_learners = []
+                    for user_email in user_email_batch:
+                        if user_email:
+                            try:
+                                user_id = User.objects.get(email=user_email).pk
+                                enterprise_customer_user, __ = models.EnterpriseCustomerUser.objects.update_or_create(
+                                    enterprise_customer=group.enterprise_customer,
+                                    user_id=user_id,
+                                    defaults={'active': True},
+                                )
+                                enterprise_learners.append(
+                                    models.EnterpriseGroupMembership(
+                                        enterprise_customer_user=enterprise_customer_user,
+                                        group=group,
+                                    )
+                                )
+                                total_existing_users_processed += 1
+                            except User.DoesNotExist:
```

Comment:
Do we need to rely on catching exceptions in the (as I understand it) un-exceptional case of adding a new user?

Reply:
I reworked the logic so this is no longer relevant